### ISSUE TITLE
[Discussion - input request] - Adding of information for assisting / improving agentic workflows in the repo / general agentic use discussion

### ISSUE THREAD

**Issue by AoifeHughes** [alan-turing-institute/ice-station-zebra#392]
### Problem / Goal

@IFenton had brought up a few weeks ago that other teams had asked about things like `agents.md` being part of the project and if that was something we'd be interested in for this project.

Initially, and currently I'm generally against the idea of an `agents.md` (I could probably be persuaded and am open to others' thoughts), as I find the most benefit in generating an agents.md per task/branch etc. and Im not sure a generalised one would be useful or if it would create a burden of keeping updated.

Having said that, I have been experimenting more with the ideas of `skills` ([for ref](https://agentskills.io/home)) in my own workflow and whilst I'm not entirely convinced on having them in the repo either, I also thought it could be something worth sharing somehow as I'm sure others' on the project also have great workflows that would be generally useful to share. 

I've put an example of a `train_a_model.md` skill below, but maybe sharing / having these for testing, running ci etc. might be useful for focusing workflows.

Again, I don't feel strongly on this and don't use them myself for this project specifically, but thought documenting a discussion around this + agentic things in general might be useful. Along the same lines it might be useful to use this issue to discuss other things like "do we want to make PR templates have an AI disclosure?" section for clarity and transparency? 

### What does good look like?

An example I wrote specifically for this discussion (its unused, please don't critique it, it's purely for example and discussion):

--- 
# Train a Model in IceNet-MP

## When to use this skill

Use when the user wants to train, evaluate, or run multistage training on models in IceNet-MP. Also use for configuring training hyperparameters, managing checkpoints, interpreting W&B results, and running evaluation jobs on trained models.

## Prerequisites

1. **IceNet-MP installed** — see `docs/src/user-guide/installation.md`
2. **Weights & Biases account** — generate an API key and authenticate:
   ```bash
   export WANDB_API_KEY=<your_api_key>
   wandb login
   ```
3. **Data available** — either pre-downloaded on HPC (Baskerville, DAWN, Isambard-AI) or downloaded locally via `uv run imp datasets create`

## 1. Create a Local Config

Create a file at `icenet_mp/config/<your-name>.local.yaml`. The base config depends on your environment:

### On Isambard-AI
```yaml
defaults:
  - base_isambardai
  - _self_
```

### On a local machine
```yaml
defaults:
  - base
  - _self_

base_path: /path/to/your/data
```

### Reusing a previous W&B run config
Download `model_config.yaml` from the W&B run page (**Files** tab) and place it at `icenet_mp/config/<your-name>.local.yaml`. Update the `defaults` block for your machine.

## 2. Run Training

```bash
uv run imp train --config-name <your-name>.local
```

### Configuring Training Hyperparameters

Training is controlled by the `train` section of your config:

```yaml
train:
  optimizer:
    lr: 1e-3
    weight_decay: 1e-4
  scheduler:
    T_max: 20
    eta_min: 1e-5
  trainer:
    max_epochs: 20
    accelerator: auto
```

### Decoder Options

**Mask type** — controls which grid cells are zeroed (land / no-ice cells):
```yaml
decoder:
  mask_type: active   # active+land | land | none
```

**Output range restriction:**
```yaml
decoder:
  restrict_range: sigmoid   # none | sigmoid | clamp | tanh
```

## Checkpoints

Checkpoints are saved after each epoch to:
```
${BASE_DIR}/training/wandb/run-<date>-<id>/checkpoints/<name>.ckpt
```

where `BASE_DIR` is the `base_path` from your local config.

## 3. Inspect Results in W&B

Training runs appear in the W&B project **train** under entity **turing-seaice**.

| Panel | What it shows |
|-------|---------------|
| `train_loss` / `validation_loss` | Loss per epoch — watch for validation loss converging or diverging from training loss. |
| `train_{metric}_mean` / `validation_{metric}_mean` | Per-epoch mean of each metric (e.g. SIE error, IceNet accuracy). |
| `{metric}_per_forecast_day` | Metric broken down by forecast lead day — useful for diagnosing skill drop at longer horizons. |

Use the W&B workspace to overlay multiple runs or group them by config keys to compare experiments.

## Running Evaluations

Evaluate a trained checkpoint:

```bash
uv run imp evaluate --config-name <your-name>.local --checkpoint PATH_TO_CHECKPOINT
```

### Enabling Visualisations

By default, all visualisations are enabled. To disable forecast plots:
```yaml
evaluate:
  callbacks:
    plotting:
      make_static_plots: false
      make_video_plots: false
      make_input_plots: false   # raw input data plots
```

Evaluation results appear in the W&B project **evaluate** under entity **turing-seaice**, with panels for static images, animated forecasts, and per-forecast-day metrics.

## Multistage Training

For `EncodeProcessDecode` models, components can be pretrained independently before final fine-tuning:
1. Train encoders in isolation
2. Train decoders in isolation
3. Freeze encoders/decoders and train the processor on fixed latent space
4. Optionally finetune the full model end-to-end

See `docs/src/how-to/train-multistage.md` for details.

## Key Files to Reference

| Resource | Path |
|----------|------|
| Training guide | `docs/src/how-to/train.md` |
| Multistage training | `docs/src/how-to/train-multistage.md` |
| Evaluation guide | `docs/src/how-to/evaluate.md` |
| Configuration reference | `docs/src/user-guide/configuration.md` |
| CLI commands | `docs/src/user-guide/commands.md` |

**Comment by AoifeHughes** [comment]
I realise the formatting is a little ugly of the example above, but it contains markdown embeded with other code samples so it would have been a faff to do nicely, I think it serves its purpose as an example piece.

**Comment by jemrobinson** [comment]
When skills are structured like this, it makes me think that they would be better organised as how-to guides in the docs (then they can be human-readable too). Do you think that would make sense? You can then add some local skills that say "read this bit of the local docs" if you want?

**Comment by AoifeHughes** [comment]
Hmm, that might actually be a really nice solution. 

The benefit to skills for an agent is not having to read them but be aware of them so there's no context issues for unrelated things. Searching it out via docs adds an extra small step. 

Saying that, it could also be a cheap solution for an agents.md file to effectively say "read the docs".

**Comment by IFenton** [comment]
I agree that making them human readable as well sounds logical

**Comment by AoifeHughes** [comment]
The main thing I'd worry about for going down the "human readable" route is that we tend to write things with a lot of prose, even technical docs, whereas AI models generally favour mode direct-firm imperatives. Trying to do both could be a solution which isn't ideal for either.

**Comment by jemrobinson** [comment]
@AoifeHughes do you want to argue for this to go into our next milestone?
